### ISSUE TITLE
Unable to use specific VueJS attribute in Shared::Field

### ISSUE THREAD

**Issue by brunto** [luckyframework/lucky#1409]
**Describe the bug**
When adding some VueJS attributes like `v-on:change` or `@change` to a `select_input()` inside a `Shared:Field` it produce an error.

**To Reproduce**
Install VueJS on a lucky application (of course :) )
Create a form with a line like this:
`mount Shared::Field, operation.some_field, &.select_input({"@change" => "onchoose($event)"}) do`

**Expected behavior**
We should have the attribute added to the select HTML element like:
`<select @change="onchoose($event)">`
We also need to have the ability to use `v-on:change`.

**Versions (please complete the following information):**
 - Lucky version: 0.25.0
 - Crystal version: 0.35.1
 - OS: MacOS BigSur

**Comment by matthewmcgarvey** [comment]
@brunto Lucky just does a catch-all for named arguments and puts them in the HTML tag. This should work

```crystal
mount Shared::Field, operation.some_field, &.select_input({"@change": "onchoose($event)"})
```

**Comment by matthewmcgarvey** [comment]
Ahh! nevermind. The error is specific to the `Lucky::TagDefaults` module (tag builder)

**Comment by matthewmcgarvey** [comment]
After looking into this, I'm not sure it's possible to solve this issue with the way the tag defaults code works right now.

The `Lucky::TagDefaults` module implements the `method_missing` macro https://github.com/luckyframework/lucky/blob/42e9b7b43356b729bf2b055b3ee9af5f2b92d768/src/lucky/tags/tag_defaults.cr#L31

It receives the intended method call and modifies it before calling the context. The problem is that calling a named arg like `select_input("@change": "onchoose($event)")` doesn't work with the method missing macro. When working with named args you can access the named args name (in this case it would be `"@change"`) but you cannot access the named arguments literal value. Instead, it gives you a variable name that will provide the value at runtime. Calling `select_input(id: "my-id")` gives a named argument with a name of `id` and a value of `id`. Using the `id` variable will allow you to access the value. That doesn't work for these weird named argument names because the variable name is invalid. `"@change": "onchoose($event)"` gives a value of `@change`. That could reference an instance variable but it most certainly won't reference the named argument's real value. It seems the only way we could make this work is to avoid macros or to provide a way to pass in something other than named arguments.

Maybe something like:

```crystal
mount Shared::Field, operation.some_field, &.select_input(custom_attrs: {"@change" => "onchoose($event)"})
```

In that way we pass on a hash instead of a named argument.

**Comment by matthewmcgarvey** [comment]
I made a forum post about this https://forum.crystal-lang.org/t/named-arguments-in-macros-with-symbolized-strings/2950

If this is not considered a bug (maybe even if it is since it would mean waiting on the next release if it's even solved quickly), we should figure out how to decorate the method call without using the method_missing macro.

**Comment by matthewmcgarvey** [comment]
https://github.com/crystal-lang/crystal/issues/10381 is related to this. The language maintainers don't like the method_missing macro and would look to review it. Whether or not that happens, I don't think there's going to be a quick turnaround on fixin gthis problem and we should consider alternatives.

I'm not a fan of adding special handling for these types of attributes since the regular html dsl supports it.

**Comment by matthewmcgarvey** [comment]
There's a PR opened to fix this issue https://github.com/crystal-lang/crystal/pull/10388

Again, I still believe we should search for an alternative to using method_missing

**Comment by matthewmcgarvey** [comment]
We should be able to fix this when Lucky supports >= Crystal 1.0
